Content.Server/Nutrition/EntitySystems/FatExtractorSystem.cs

```diff
@@ -0,0 +1,169 @@
+    {
+        args.Handled = true;
+        args.Repeatable = false;
+    }
+
+    private void OnClosed(EntityUid uid, FatExtractorComponent component, ref StorageAfterCloseEvent args)
+    {
+        StartProcessing(uid, component);
+    }
+
+    private void OnOpen(EntityUid uid, FatExtractorComponent component, ref StorageAfterOpenEvent args)
+    {
+        StopProcessing(uid, component);
+    }
+
+    private void OnPowerChanged(EntityUid uid, ActiveFatExtractorComponent component, ref PowerChangedEvent args)
+    {
+        if (!args.Powered)
+            StopProcessing(uid, active: component);
+    }
+
+    public void StartProcessing(EntityUid uid, FatExtractorComponent? component = null, EntityStorageComponent? storage = null)
+    {
+        if (!Resolve(uid, ref component, ref storage))
+            return;
+
+        if (HasComp<ActiveFatExtractorComponent>(uid))
+            return;
+
+        if (!this.IsPowered(uid, EntityManager))
+            return;
+
+        if (!TryGetValidOccupant(uid, out _, component, storage))
+            return;
+
+        EnsureComp<ActiveFatExtractorComponent>(uid);
+        _appearance.SetData(uid, FatExtractorVisuals.Processing, true);
+        component.Stream = _audio.PlayPvs(component.ProcessSound, uid);
+        component.NextUpdate = _timing.CurTime + component.UpdateTime;
+    }
+
+    public void StopProcessing(EntityUid uid, FatExtractorComponent? component = null, ActiveFatExtractorComponent? active = null)
+    {
+        if (!Resolve(uid, ref component, ref active, false))
+            return;
+
+        RemComp(uid, active);
+        _appearance.SetData(uid, FatExtractorVisuals.Processing, false);
+        component.Stream?.Stop();
+    }
+
+    public bool TryGetValidOccupant(EntityUid uid, [NotNullWhen(true)] out EntityUid? occupant, FatExtractorComponent? component = null, EntityStorageComponent? storage = null)
+    {
+        occupant = null;
+        if (!Resolve(uid, ref component, ref storage))
+            return false;
+
+        occupant = storage.Contents.ContainedEntities.FirstOrDefault();
+
+        if (!TryComp<HungerComponent>(occupant, out var hunger))
+            return false;
+
+        if (hunger.CurrentHunger < component.NutritionPerSecond)
+            return false;
+
+        if (hunger.CurrentThreshold < component.MinHungerThreshold && !HasComp<EmaggedComponent>(uid))
+            return false;
+
+        return true;
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<ActiveFatExtractorComponent, FatExtractorComponent, EntityStorageComponent>();
+        while (query.MoveNext(out var uid, out var active, out var fat, out var storage))
+        {
+            if (_timing.CurTime < fat.NextUpdate)
+                continue;
+            fat.NextUpdate = _timing.CurTime + fat.UpdateTime;
```

Comment:
Technically you should do += UpdateTime as you will get rounding issues on a long enough timespan, especially if the tickrate ever changes, because any remaining time gets eaten up.

This just means you need to set the time to current time on init if it's < than current time.